packages/ui/src/components/form-input/index.ts

```diff
@@ -17,7 +17,17 @@ const FormInput = {
     checked: field.value,
     onCheckedChange: field.onChange
   })),
+  Select: withForm(Select, ({ field }) => ({
+    onSelectedChange: field.onChange
+  })) as <T = string>(
+    props: FormSelectProps<T> & { ref?: React.ForwardedRef<HTMLButtonElement> }
```

Comment:
can we remove this type casting and make relevant changes in `Select` component? `Select` should still return `ForwardRefExoticComponent`